PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -881,7 +887,7 @@ LoadBootImages (
     ContainerImage = &BootImage[Index].ContainerImage;
     if ((ContainerImage->Indicate == '!') && (ContainerImage->BackSlash == '/')) {
       Status = GetBootImageFromIfwiContainer (OsBootOption, LoadedImage);
-    } else if (BootImage[Index].LbaImage.Valid == 1) {
+    } else if (BootImage[Index].LbaImage.Valid == 1 || OsBootOption->DevType == OsBootDeviceMemory) {
```

Comment:
We don't need take memory device as a special device. Valid bit should be set if booting using Lba.

Reply:
Updated